src/distribute.ts

```diff
@@ -0,0 +1,135 @@
+import { writeFileSync } from 'fs';
+import { arweave, readInitialOutputFile } from './common';
+import type { Rewards } from './inferno_types';
+import Transaction from 'arweave/node/lib/transaction';
+
+export type TransactionToDistribute = {
+	id: string;
+	tx: Transaction;
+	target: string;
+	qty: number;
+	rankPosition: number;
+};
+
+// Get the key file
+const keyfile = process.env.KEYFILE ? JSON.parse(process.env.KEYFILE) : undefined;
+
+export async function distributeTokens(confirm: boolean) {
+	const data = readInitialOutputFile();
+	const weeklyRank = data.ranks.weekly;
+
+	if (!keyfile) {
+		console.log("There's no keyfile to create transactions");
+		return;
+	}
+
+	if (!weeklyRank.hasReachedMinimumGroupEffort) {
+		console.log("The group didn't reach the minimum effort this week");
+		return;
+	}
+
+	if (confirm) {
+		console.log('========= CREATING TRANSACTIONS =========');
+	} else {
+		console.log('DRY-RUN: THE SCRIPT IS NOT POSTING TRANSACTIONS');
+	}
+
+	const addresses = weeklyRank.groupEffortRewards.filter((wallet) => wallet.rewards > 0);
+	const transactions = await createTransactions(addresses);
+
+	if (confirm) {
+		console.log('========= SENDING TRANSACTIONS =========');
+	}
+
+	const transactionsToReport: { id: string; qty: number; rankPosition: number }[] = [];
+
+	for (const transaction of transactions) {
+		if (!confirm) {
+			console.log(transaction);
+			return;
+		}
+
+		const transactionSent = await sendTransaction(transaction);
+		if (transactionSent) {
+			writeFileSync(`./distribution/${transaction.id}`, JSON.stringify(transaction, null, '\t'));
+			transactionsToReport.push({
+				id: transaction.id,
+				qty: transaction.qty,
+				rankPosition: transaction.rankPosition
+			});
+		}
+	}
+
+	console.log('Writing report to distribution/report.json');
+	writeFileSync('./distribution/report.json', JSON.stringify(transactionsToReport, null, '\t'));
+}
+
+export async function createTransactions(wallets: Rewards): Promise<TransactionToDistribute[]> {
```

Comment:
Having `createTransactions` in the global namespace is kind of vague. Consider `createTokenDistributionTxs` or something specific.

Reply:
https://github.com/ardriveapp/inferno-backend/pull/24/commits/8d62c38466119c0bc089bae0c3fdc65c670d2a82